src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ReuploadFailedEnvelopeTask.java

```diff
@@ -55,29 +56,37 @@ public FailedDocUploadProcessor getProcessor() {
     @SchedulerLock(name = "re-upload-failures")
     @Scheduled(fixedDelayString = "${scheduling.task.reupload.delay}")
     public void processUploadFailures() throws InterruptedException {
-        ExecutorService executorService = Executors.newFixedThreadPool(
-            accessMapping.size(),
-            r -> new Thread(r, "BSP-REUPLOAD")
-        );
-        CompletionService<Void> completionService = new ExecutorCompletionService<>(executorService);
+        ExecutorService executorService = null;
 
-        accessMapping
-            .stream()
-            .map(Mapping::getJurisdiction)
-            .forEach(jurisdiction -> {
-                FailedDocUploadProcessor processor = getProcessor();
+        try {
+            executorService = Executors.newFixedThreadPool(
+                accessMapping.size(),
+                new ThreadFactoryBuilder().setNameFormat("BSP-REUPLOAD-%d").build()
+            );
 
-                completionService.submit(() -> {
-                    log.info("Processing failed documents for jurisdiction {}", jurisdiction);
+            CompletionService<Void> completionService = new ExecutorCompletionService<>(executorService);
```

Comment:
so if it's `null` it'll throw an error or just use default construct? if such exists. eitherway it's good - early failure ;)

Reply:
At this point executorService won't be null. The diff isn't great, but what I've done is I wrapped the whole thing in try-finally and moved the `shutdown` call to finally clause.

Reply:
I meant for completion service. but eitherway - summing up in `finally` block sorts things out